gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -299,6 +335,44 @@ public Boolean getKeepAliveWithoutCalls() {
       return keepAliveWithoutCalls;
     }
 
+    /**
+     * Number of underlying grpc channels to open. Calls will be load balanced round robin across
+     * them.
+     */
+    public int getPoolSize() {
+      return poolSize;
+    }
+
+    /**
+     * Number of underlying grpc channels to open. Calls will be load balanced round robin across
+     * them
+     */
+    public Builder setPoolSize(int poolSize) {
+      Preconditions.checkArgument(poolSize > 0, "Pool size must be positive");
+      this.poolSize = poolSize;
+      return this;
+    }
```

Comment:
Nit: I usually like `this.poolSize = checkArgument()`, but up to you.

Reply:
checkArgument doesn't return a value...I think you meant checkNotNull

Reply:
Derp, you're right. Wasn't thinking.